<launch>
    <!-- Camera RBG Image Raw topic -->
    <arg name ="input_image_topic"      default="/camera/camera/color/image_raw"/>

    <!-- Select the camera base frame -->
    <arg name ="base_frame_name"        default="camera_link"/>
    <!-- Select the cloud topic name -->
    <arg name ="cloud_topic_name"       default="/camera/depth/points"/>
    <!-- Select the camera_info topic name -->
    <arg name ="camera_info_topic_name" default="/camera/rgb/camera_info"/>

    <!-- Enable 2D Pose detection (true) -->
    <arg name ="pose_2d_detect"         default="true"/>
    <!-- Show 2D Pose result image (true) -->
    <arg name ="pose_2d_img_show"       default="true"/>
    <!-- Publish 2D Pose result image (true) -->
    <arg name ="pose_2d_img_pub"        default="true"/>
    <!-- Show 2D Pose result as log in terminal (true) -->
    <arg name ="pose_2d_log_show"       default="false"/>
    
    <!-- Enable 3D Pose detection (true) -->
    <arg name ="pose_3d_detect"         default="true"/>
    <!-- Publish 3D Pose result as topic (true) -->
    <arg name ="pose_3d_topic_pub"      default="false"/>
    <!-- Broadcast 3D Pose result as TF (true) -->
    <arg name ="pose_3d_tf_pub"         default="true"/>
    <!-- Show 3D Pose result as log in terminal (true) -->
    <arg name ="pose_3d_log_show"       default="false"/>

    <arg name ="checkpoint_path"        default="$(env COLCON_WORKSPACE)/lightweight_human_pose_estimation/lightweight_human_pose_estimation/script/checkpoints/checkpoint_iter_370000.pth"/>
    <arg name ="height_size"            default="256"/>
    <arg name ="cpu"                    default="false"/>
    <arg name ="track"                  default="true"/>
    <arg name ="smooth"                 default="true"/>

    <node pkg = "lightweight_human_pose_estimation" exec="human_pose_2d.py" output="screen">
        <param name="checkpoint_path"   value="$(var checkpoint_path)"/>
        <param name="height_size"       value="$(var height_size)"/>
        <param name="cpu"               value="$(var cpu)"/>
        <param name="track"             value="$(var track)"/>
        <param name="smooth"            value="$(var smooth)"/>

        <param name="input_image_topic" value="$(var input_image_topic)"/>

        <param name="pose_2d_detect"    value="$(var pose_2d_detect)"/>
        <param name="pose_2d_img_show"  value="$(var pose_2d_img_show)"/>
        <param name="pose_2d_img_pub"   value="$(var pose_2d_img_pub)"/>
        <param name="pose_2d_log_show"  value="$(var pose_2d_log_show)"/>
    </node>

    <!-- <node name="human_pose_3d" pkg="lightweight_human_pose_estimation" type="human_pose_3d" respawn="false" output="screen">
        <param name="base_frame_name"    value="$(arg base_frame_name)"/>
        <param name="cloud_topic_name"   value="$(arg cloud_topic_name)"/>
        <param name="camera_info_topic_name" value="$(arg camera_info_topic_name)"/>

        <param name="pose_3d_detect"    type="bool" value="$(arg pose_3d_detect)"/>
        <param name="pose_3d_topic_pub" type="bool" value="$(arg pose_3d_topic_pub)"/>
        <param name="pose_3d_tf_pub"    type="bool" value="$(arg pose_3d_tf_pub)"/>
        <param name="pose_3d_log_show"  type="bool" value="$(arg pose_3d_log_show)"/>
    </node> -->
</launch>